src/main/java/org/reaktivity/ry/internal/RyMain.java

```diff
@@ -15,68 +15,178 @@
 
 public final class RyMain
 {
     private static final String RY_EXECUTABLE_NAME = "ry";
-    //private static final String RY_DEPENDENCIES_NAME = "ry.deps";
+
+    private static final String DEPENDENCY_FILENAME = "ry.deps";
+    private static final String DEPENDENCY_LOCK_FILENAME = String.format("%s.lock", DEPENDENCY_FILENAME);
+
+    private static final String PROPERTY_DEPENDENCIES = "dependencies";
+
+    private static final String CACHE_DIR = String.format("%s/.ry", System.getProperty("user.home"));
+
+    private static final String DEFAULT_GROUP_ID = "org.reaktivity";
+
+    private static Map<String, String> dependencies = new LinkedHashMap<>(); // TODO Done correctly?
+    
+    private static List<URL> artifactOrigins = new LinkedList<>(); // TODO Done correctly?
 
     public static void main(
         String[] args)
     {
+        DefaultMessageLogger logger = new DefaultMessageLogger(Message.MSG_INFO);
+
+        ResolveOptions options = new ResolveOptions();
+        options.setLog(ResolveOptions.LOG_DOWNLOAD_ONLY);
+
         final CliBuilder<Runnable> builder = Cli.<Runnable>builder(RY_EXECUTABLE_NAME)
                 .withDefaultCommand(Help.class)
                 .withCommand(Help.class);
 
-        ClassLoader loader = Thread.currentThread().getContextClassLoader();
-        for (RyCommandSpi service : ServiceLoader.load(RyCommandSpi.class, loader))
+        try
         {
-            service.mixin(builder);
+            readDepsFile();
+
+            boolean resolved = resolveDependencies(options);
+            if (!resolved)
+            {
+                throw new Exception("Dependencies failed to resolve");
+            }
+
+            URLClassLoader loader  = new URLClassLoader(
+                (URL[]) artifactOrigins.toArray(new URL[artifactOrigins.size()]),
+                Thread.currentThread().getContextClassLoader());
+            for (RyCommandSpi service : ServiceLoader.load(RyCommandSpi.class, loader))
+            {
+                service.mixin(builder);
+            }
+        }
+        catch (Exception ex)
+        {
+            logger.error(String.format("Error: %s", ex.getMessage()));
+        }
+        finally
+        {
+            logger.sumupProblems();
         }
 
         final Cli<Runnable> parser = builder.build();
         final Runnable command = parser.parse(args);
-
+        
         command.run();
     }
 
-    /*
-    public static ClassLoader initClassLoader()
+    private static void readDepsFile() throws IOException
     {
-        URL[] deps;
+        JsonElement deps = null;
+        File depsFile = new File(DEPENDENCY_LOCK_FILENAME);
+        if (!depsFile.exists())
+        {
+            depsFile = new File(DEPENDENCY_FILENAME);
+            if (!depsFile.exists())
+            {
+                throw new FileNotFoundException(
+                    String.format("Cannot find %s or %s", DEPENDENCY_LOCK_FILENAME, DEPENDENCY_FILENAME));
+            }
+        }
```

Comment:
Suggest hoisting determination of which file to parse into separate method, then pass `File` to `readDepsFile(...)`.